python/gui/MaaS/static/common/js/domain.js

```diff
@@ -0,0 +1,206 @@
+                }
+            }
+        }
+    );
+}
+
+function insertOptionInOrder(option, newParentSelect) {
+    var next_i, i = 0, next_size = 200;
+
+    next_i = i + next_size;
+    while (next_i < newParentSelect.options.length) {
+        if (parseInt(option.value) < parseInt(newParentSelect.options[next_i].value)) {
+            break;
+        }
+        else {
+            i = next_i;
+            next_i = i + next_size;
+        }
+    }
+
+    for (i; i < newParentSelect.options.length; i++) {
+        if (parseInt(option.value) < parseInt(newParentSelect.options[i].value)) {
+            newParentSelect.options.add(option, newParentSelect.options[i]);
+            return;
+        }
+    }
+    newParentSelect.appendChild(option);
+}
+
+function addDomainChoicesOption(values) {
+    var select = document.getElementById('domainChoices');
+    for (var i = 0; i < values.length; i++) {
+        var option = document.createElement('option');
+        option.value = values[i].substring(4);
+        option.innerHTML = values[i];
+        insertOptionInOrder(option, select);
+    }
+}
+
+function controlSelectAdd() {
+    var choices = document.getElementById('domainChoices'),
+        selected = document.getElementById('domainSelections');
+    for (var i = choices.options.length - 1; i >=0; i--) {
+        var opt = choices.options[i];
+        if (opt.selected) {
+            opt.selected = false;
+            choices.removeChild(opt);
+            insertOptionInOrder(opt, selected);
+        }
+    }
+}
+
+function controlSelectRemove() {
+    var choices = document.getElementById('domainChoices'),
+        selected = document.getElementById('domainSelections'),
+        i,
+        opt;
+    for (i = selected.options.length - 1; i >=0; i--) {
+        opt = selected.options[i];
+        if (opt.selected) {
+            opt.selected = false;
+            selected.removeChild(opt);
+            insertOptionInOrder(opt, choices);
+        }
+    }
+}
+
+function controlSelectAll() {
+    var choices = document.getElementById('domainChoices'),
+        selected = document.getElementById('domainSelections'),
+        i,
+        opt;
+    for (i = choices.options.length - 1; i >= 0 ; i--) {
+        opt = choices.options[i];
+        if (opt.selected) {
+            opt.selected = false;
+        }
+        choices.removeChild(opt);
+        insertOptionInOrder(opt, selected);
+    }
+}
+
+function controlSelectClear() {
+    var choices = document.getElementById('domainChoices'),
+        selected = document.getElementById('domainSelections'),
+        i,
+        opt;
+    for (i = selected.options.length - 1; i >= 0 ; i--) {
+        opt = selected.options[i];
+        if (opt.selected) {
+            opt.selected = false;
+        }
+        selected.removeChild(opt);
+        insertOptionInOrder(opt, choices);
+    }
+}
+
+function loadFabricDomain(event) {
+    var name = $("#fabric-selector").val(),
```

Comment:
Use `let` instead of `var` and try to keep one declaration per statement.

```javascript
let name = ...;
let type = ...;
let catLists = ...;
```

Looks heavy, but the cognitive load is lower and it makes it easier to edit later on.